### ISSUE TITLE
[FR] Drop Python 3.5 support in MLflow 1.15.0

### ISSUE THREAD

**Issue by smurching** [mlflow/mlflow#3984]
## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [x] No. I cannot contribute this feature at this time -> happy to shepherd a contribution

## Proposal Summary
Python 3.5 is EOL, so we should drop support for it in upcoming releases, i.e. as soon as in the upcoming (1.14.0) MLflow release. This means the MLflow 1.13.x release series will be the last to support Python 3.5. Similar to the proposal for dropping Python 2 support in in https://github.com/mlflow/mlflow/issues/2522, we may consider backporting critical security fixes and releasing 1.13.x patch releases on a per-case basis, but otherwise will no longer support Python 3.5.

## Motivation
Dropping Python 3.5 support will reduce maintenance burden & speed up MLflow CI (as we need to test against fewer Python versions)

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dmatrix** [comment]
@smurching @harupy Shouldn't this be assigned to us?

**Comment by harupy** [comment]
@dmatrix Yes. This is more like a notification to the community.

**Comment by harupy** [comment]
Here's some stats:


| category | percent |  downloads |
| -------- | ------: | ---------: |
| 3.7      |  81.48% | 10,774,222 |
| 3.6      |  10.92% |  1,444,153 |
| 3.5      |   3.42% |    452,363 |
| 3.8      |   1.93% |    254,723 |
| 2.7      |   1.49% |    196,684 |
| null     |   0.73% |     96,172 |
| 3.9      |   0.04% |      5,025 |
| 3.4      |   0.00% |         19 |
| 2.6      |   0.00% |          1 |
| 3.3      |   0.00% |          1 |
| Total    |         | 13,223,363 |

Date range: 2020-07-01 - 2020-12-31

Source:

```
pip install -U pypistats
pypistats python_minor mlflow --start-date 2020-07 --end-date 2020-12
```

**Comment by harupy** [comment]
What we need to is:

1. Update `python_requires` to `3.6`:

https://github.com/mlflow/mlflow/blob/d977e42e710a268d08e65f6c7616d8c8f9e170e2/setup.py#L114


2. Remove `test_no_f_strings.py`:

https://github.com/mlflow/mlflow/blob/8e74cdd108ff324175e38d621b12d9b2aced05bd/tests/test_no_f_strings.py

**Comment by harupy** [comment]
I just made an announced on Slack: https://mlflow-users.slack.com/archives/CBFQ95Y2G/p1611193439002500

**Comment by harupy** [comment]
Addressed via #4179
